### DISCUSSION TITLE
Kubernetes IP Restriction

### DISCUSSION THREAD

**Discussion by rlawrie-mears** [datalust/seq-tickets#1847] [Help]
Hi,

We are planning to run SEQ in Kubernetes (Linux), and are hoping to IP restrict the portal while leaving the ingestion endpoint open (secured via API Key). Do you have any documentation to explain how to do this?

Thanks,
Rob

**Comment by nblumhardt** [comment]
Hi @rlawrie-mears, thanks for getting in touch!

The Seq container configures ports 5341 and 45341 (if SSL is terminated in the container) as ingestion-only ports, you can expose these to all IPs, while IP-locking traffic to the remaining (API/UI) ports.

Because of the potential for ports to be rerouted through ingress etc. it pays to test first, by attempting to open the ingestion port route in a web browser - you should see 403/Forbidden for any non-ingestion HTTP request made there.

If you need any more details, please just let me know.

Best regards,
Nick

**Reply by rlawrie-mears** [reply]
Thanks for the timely response Nick,

Our SSL is terminated in the ingress controller, is container based termination the recommended method? Anyway to get this to work while still terminating in the ingress controller? I'm fairly new to Kubernetes so still learning best practice etc!

Thanks
Rob

**Reply by nblumhardt** [reply]
Hi Rob; sorry about my unclear reply, there! 5341 is always available as an ingestion port (plain HTTP). It's only 45341 that requires SSL termination at the container.

I think what you need is to map port 443 at the ingress to 80 on the container, so that UI/API traffic is SSL to the ingress, and clear to the container.

Then, also at the ingress, map 45341 (with SSL termination) to port 5341 on the container, so that you get ingestion only traffic to 45341 at the ingress, forwarded in the clear to the ingestion-only port on the container.

Let me know if this sounds like it all hangs together!

Best regards,
Nick
